compiler/resolution/functionResolution.cpp

```diff
@@ -8263,6 +8263,31 @@ buildRuntimeTypeInfo(FnSymbol* fn) {
 }
 
 
+static void lowerAutoDestroyRuntimeTypes() {
+  forv_Vec(CallExpr, call, gCallExprs)
+    if (call->inTree() && call->isPrimitive(PRIM_AUTO_DESTROY_RUNTIME_TYPE)) {
+     if (SymExpr* rttSE = toSymExpr(call->get(1)))
+      // toAggregateType() filters out calls in unresolved generic functions.
+      if (AggregateType* rttAG = toAggregateType(rttSE->symbol()->type))
+       if (rttAG->symbol->hasFlag(FLAG_RUNTIME_TYPE_VALUE))
+        if (Symbol* domField = rttAG->getField("dom", false))
+         if (FnSymbol* destroyFn = autoDestroyMap.get(domField->getValType()))
+          {
+           // Invoke destroyFn on rttSE->dom.
+           INT_ASSERT(call->getStmtExpr() == call);
+           SET_LINENO(call);
+           VarSymbol* domTemp = newTemp("domTemp", domField->getValType());
+           call->insertBefore(new DefExpr(domTemp));
+           call->insertBefore("'move'(%S,'.v'(%E,%S))", domTemp,
+                              rttSE->remove(), domField);
+           call->insertBefore(new CallExpr(destroyFn, domTemp));
+          }
+     // Otherwise this call is a no-op. Remove it.
+     call->remove();
+    }
+}
```

Comment:
It seems to me that callDestructors would be a better place for this code. That's assuming it can't be in preFold or postFold, (presumably because the runtime types don't exist yet?).

Reply:
Also this code can't handle runtime types for domains (which store distributions). At least it needs a comment that it would need updating.